x-pack/filebeat/input/salesforce/input.go

```diff
@@ -341,19 +357,23 @@ func (s *salesforceInput) RunEventLogFile() error {
 
 			resp, err := s.sfdcConfig.Client.Do(req)
 			if err != nil {
-				return err
+				return fmt.Errorf("error fetching log file: %w", err)
 			}
 
-			body, err := io.ReadAll(resp.Body)
-			if err != nil {
+			if resp.StatusCode != http.StatusOK {
 				resp.Body.Close()
-				return err
+				return fmt.Errorf("unexpected status code %d for log file", resp.StatusCode)
 			}
```

Comment:
most important change. only resp with 200 status code should be allowed to go forward.